presto-hive/src/main/java/io/prestosql/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -375,6 +376,7 @@ private void invokeNoMoreSplitsIfNecessary()
                         partitionMatchSupplier,
                         partition.getTableToPartitionMapping(),
                         Optional.empty(),
+                        getMaxSplitSize(session),
```

Comment:
These should be `getMaxInitialSplitSize()`, otherwise we won't split files that are between the initial size and max size (default in the 32MB to 64MB range).

Reply:
I can see the argument either way here actually. It's true that this would bypass the initial split size logic, but when splits are *smaller* than 64MB do we really want to spend time trying to split them into parts or do we want to bias towards avoiding the isSplittable check?

I've been leaning towards this iteration, especially since there will only be a difference for the first X splits generated but I could be convinced to go the other way on it.

Reply:
The initial split size is about increasing parallelism for small tables, so I would say we do what that.

Reply:
It's worth noting that this interacts with the behavior change in https://github.com/prestosql/presto/pull/4485. When producing initial splits from files between maxInitialSplitSize and maxSplitSize (32MB and 64MB respectively by default):
- If we choose max initial split size as the threshold, those will yield initial splits in the range of 16-32MB because they will be considered splitable and then split into equal portions.
- If we choose max split size as the threshold, files those will yield a single split between 32-64MB past which point this difference is not especially material because standard split generation logic holds.

Reply:
After some offline discussion, I'm in the camp now where I think that max initial split size is the right trade off here. Since the code already reflects that, this is ready for re-review.